specs/_features/eip7594/p2p-interface.md

```diff
@@ -148,7 +163,7 @@ The *type* of the payload of this topic is `DataColumnSidecar`.
 
 The following validations MUST pass before forwarding the `sidecar: DataColumnSidecar` on the network, assuming the alias `block_header = sidecar.signed_block_header.message`:
 
-- _[REJECT]_ The sidecar's index is consistent with `NUMBER_OF_COLUMNS` -- i.e. `sidecar.index < NUMBER_OF_COLUMNS`.
+- _[REJECT]_ The sidecar is valid as verified by `verify_data_column_sidecar(sidecar)`.
```

Comment:
these validations are generally structured from least expensive to most expensive

I think the order you have is pretty good, but may be worth another look just to make sure it would be one of the easiest things to fail first on if a peer fed us bad data

Reply:
Good idea. I moved the index check in `verify_data_column_sidecar` to the top. But yeah, all of the checks in this functions are pretty simple & should be done before anything else.